source/util/FileUtils.cpp

```diff
@@ -296,6 +303,35 @@ bool FileUtils::CreateDirectoryWithPermissions(const char *dirPath, mode_t permi
     return false;
 }
 
+bool FileUtils::DirectoryExists(const std::string &dirPath)
+{
+    struct stat dirInfo;
+    if (stat(dirPath.c_str(), &dirInfo) != 0)
```

Comment:
I think we should extract the expanded absolute path of the directory before calling `stat` on the directory.

Reply:
Done.